aten/src/ATen/core/ivalue.cpp

```diff
@@ -249,6 +249,41 @@ void IValue::dump() const {
   std::cout << *this << "\n";
 }
 
+IValue IValue::deepcopy() const {
+  switch(tag) {
+    case IValue::Tag::Tensor:
+      return IValue(toTensor().clone());
+    case IValue::Tag::Tuple: {
+      std::vector<IValue> copied_elements;
+      for (const auto& e : toTuple()->elements()) {
+        copied_elements.push_back(e.deepcopy());
+      }
+      return IValue(ivalue::Tuple::create(copied_elements));
+    }
+    case IValue::Tag::String:
+      return IValue(ivalue::ConstantString::create(toString()->string()));
+    case IValue::Tag::GenericList:
+      return IValue(toList().copy());
```

Comment:
This shallow copies the list, so this won't deep copy its members.

Reply:
looks like the comments are wrong here: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/List.h#L239-L244, cc @smessmer